uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -574,15 +627,42 @@ define([
       vm.pagination.pageChanged();
     }
 
+    /**
+     * Performs a set of manipulations to prepare all data for calculation and
+     * then triggers the calculation itself if it can be calculated
+     *
+     * @return {Promise}
+     */
+    function attemptCalculateBalanceChange () {
+      updateOpeningBalance();
+      setHoursDeductionsToRequest();
+      setMinMaxDatesToUI();
+      setDatesToRequest();
+      toggleBalance();
```

Comment:
can you explain this to me in this context? I can't understand if it's showing or hiding the balance

Reply:
It depends if the balance can be calculated.

Reply:
can you please explain this in more details? you really haven't told me anything

Reply:
Apologies, it is simply a helper function that hides/shows the balance breakdown depending on can it be calculated or not.

```js
/**
 * Shows or hides the balance breakdown depending on various conditions
 */
function toggleBalance () {
  vm.uiOptions.showBalance = vm.canCalculateChange();
}
```